Replace Grunt with Gulp

dcalhoun opened the pull request:
This replaces Grunt with Gulp for all tasks and asset compilation. Feature parity should be upheld. Resolves #52. Fixes #60.

@derekblank @maxbeizer I'd greatly appreciate eyes on this code and also a view test runs generating new projects with this version of Playbook. If you find a feature missing or a bug, please let me know here.

maxbeizer commented:
@dcalhoun initial tests look good. I'm going to keep trying things to see if I can break it.

maxbeizer commented on README.md:
```diff
@@ -38,17 +37,12 @@ yo playbook
-##### grunt deploy
-Deploy the production version of the source to [GitHub Pages](http://pages.github.com/). This ability is configurable during Playbook generation.
-
-If you choose to utilize `grunt-build-control` to deploy a GitHub Pages user/organization site, your source must live in a branch other than `master`. GitHub Pages [user/organization sites](https://help.github.com/articles/user-organization-and-project-pages#project-pages) serve the files found in the `master` branch to the browser.
```
Is there a reason behind removing this paragraph? I like the cleaner look of the shortened description, but the tradeoff is clarity for users. I would imagine one argument for removal is that this relates more to GH-pages functionality instead of generator-playbook's.

I'm on the fence about removing this, however, as the deploy task seems geared towards GH-pages. Thoughts?

maxbeizer commented on app/templates/gulpfile.js:
```diff
@@ -0,0 +1,154 @@
+    browserSync = require('browser-sync'),
+    runSequence = require('run-sequence'),
+    del = require('del'),
+    neat = require('node-neat').includePaths;
```
This style of variable assignment is different than in `index.js`.

This is pretty nitpicky, but I think we should have consistency in style. FWIW, I vote for the other style.

maxbeizer commented on app/templates/gulpfile.js:
```diff
@@ -0,0 +1,154 @@
+
+  var spawn  = require('child_process').spawn,
+      jekyll = spawn('jekyll', ['build', '-q', '--config', config, '-s', paths.app, '-d', dest], { stdio: 'inherit' });
+
```
See comment from line nine of this file. This style of variable assignment is different than the one in index.js.

dcalhoun commented:
@maxbeizer I updated the `index.js` file to match the syntax style found in the `gulpfile.js` file. I also added the information about GitHub Pages back to the README. Thanks for the feedback.

dcalhoun commented:
@derekblank do you have any thoughts or would you like more time to review? If not, I may merge this and work towards releasing 2.0 soon.
